CodeEdit/Features/Settings/SettingsView.swift

```diff
@@ -7,50 +7,123 @@
+        _ settingsStruct: Any,
+        parent: SettingsPage,
+        prePages: [SettingsPage]
+    ) -> [SettingsPage] {
+        var pages: [SettingsPage] = prePages
+        pages.append(parent)
+
+        for setting in SettingsData().propertiesOf(settingsStruct) {
+            pages.append(
+                SettingsPage(
+                    parent.name,
+                    baseColor: parent.baseColor,
+                    icon: parent.icon,
+                    isSetting: true,
+                    displayName: setting.nameString
+                )
+            )
+        }
+
+        return pages
+    }
+
+    /// Creates all the neccessary pages
+    private func populatePages() -> [SettingsPage] {
+        var pages = [SettingsPage]()
+        let settingsData = SettingsData()
+
+        let generalSettings = SettingsPage(.general, baseColor: .gray, icon: .system("gear"))
+        pages = createPageAndSettings(settingsData.general, parent: generalSettings, prePages: pages)
+
+        let accountsSettings = SettingsPage(.accounts, baseColor: .blue, icon: .system("at"))
+        pages = createPageAndSettings(settingsData.accounts, parent: accountsSettings, prePages: pages)
+
+        let themesSettings = SettingsPage(.theme, baseColor: .pink, icon: .system("paintbrush.fill"))
+        pages = createPageAndSettings(settingsData.theme, parent: themesSettings, prePages: pages)
+
+        let textEditingSettings = SettingsPage(.textEditing, baseColor: .blue, icon: .system("pencil.line"))
+        pages = createPageAndSettings(settingsData.textEditing, parent: textEditingSettings, prePages: pages)
+
+        let terminalSettings = SettingsPage(.terminal, baseColor: .blue, icon: .system("terminal.fill"))
+        pages = createPageAndSettings(settingsData.terminal, parent: terminalSettings, prePages: pages)
+
+        let sourceControlSettings = SettingsPage(.sourceControl, baseColor: .blue, icon: .symbol("vault"))
+        pages = createPageAndSettings(settingsData.sourceControl, parent: sourceControlSettings, prePages: pages)
+
+        let locationsSettings = SettingsPage(.location, baseColor: .green, icon: .system("externaldrive.fill"))
+        pages = createPageAndSettings(settingsData.sourceControl, parent: locationsSettings, prePages: pages)
+
+        let featureFlags = SettingsPage(.featureFlags, baseColor: .cyan, icon: .system("flag.2.crossed.fill"))
+        pages = createPageAndSettings(settingsData.featureFlags, parent: featureFlags, prePages: pages)
+
+        return pages
+    }
+
     var body: some View {
+        let pages: [SettingsPage] = populatePages()
         NavigationSplitView {
             List(selection: $selectedPage) {
                 Section {
-                    ForEach(Self.pages) { item in
-                        if searchText.isEmpty || item.name.rawValue.lowercased().contains(searchText.lowercased()) {
-                            SettingsPageView(item)
+                    ForEach(pages) { page in
```

Comment:
If you use the `searchKeys` method suggested above, this could be simplified. Just use the old `ForEach` when the search is empty. And use an array of `SettingsSearchResult`s to represent the search results like so:
```swift
func searchResults() -> [SettingsSearchResult] {
    // Loop through each setting page, filtering each searchKey.
}
```
You'll have to modify `SettingsSearchResult` to contain a settings page and an array of strings to represent the search matches.

Then the view would just be a ForEach over each result with another ForEach inside that for all the matching keys.
```swift
if searchString.isEmpty {
    ForEach(Self.pages) // ...
} else {
    let results = searchResults()
    ForEach(results) { (page, keys) in
        ForEach(keys) // ...
    }
}
```